Remove support for Julia <1.6

sostock opened the pull request:
The tests that are added in #551 reveal that some things don’t work correctly on Julia 1.0.

Julia 1.6 is the LTS release (since the release of Julia 1.7.0 on 2021-11-30), so support for older versions can be removed.

giordano commented:
Should we make a release before merging this?  There are some changes since last version

sostock commented:
I am _slightly_ in favor of releasing a new version before merging this, because then people on Julia 1.0 still get #530 (which one could call a bugfix). The other changes since v1.11.0 are feature additions, so IMO it doesn’t make a difference whether they are released for Julia 1.0 or not. What do you think about this?

giordano commented:
Sounds good to me.  I just noticed that in #533 I put some `if isdefined` fences instead of `if VERSION`, those should all be true for Julia >= v1.6, so the fences can be safely removed

sostock commented:
> I just noticed that in #533 I put some `if isdefined` fences instead of `if VERSION`, those should all be true for Julia >= v1.6, so the fences can be safely removed in this PR

I just removed them.

sostock commented:
Superseded by #804
